hack/lib/util/misc.sh

```diff
@@ -107,4 +107,24 @@ function os::util::format_seconds() {
 
 	printf '%02dh %02dm %02ds' "${hours}" "${minutes}" "${seconds}"
 }
-readonly -f os::util::format_seconds
\ No newline at end of file
+readonly -f os::util::format_seconds
+
+# os::util::find-go-binary locates a go install binary in GOPATH directories
+# Globals:
+#  - 1: GOPATH
+# Arguments:
+#  - 1: the go-binary to find
+# Return:
+#  None
+function os::util::find-go-binary() {
+  local binary_name="$1"
+
+  IFS=":"
+  for part in $GOPATH; do
+    if [[ -f "$part/bin/$binary_name" && -x "${part}/bin/${binary_name}" ]]; then
+      echo $part
```

Comment:
Since the method is finding the binary, I think it makes more sense for it to return the full name of the binary, not the `$GOPATH` under which it can be found.